ion/bitstream.go

```diff
@@ -660,14 +660,14 @@ func (b *bitstream) ReadBytes() ([]byte, error) {
 		panic("not a lob")
 	}
 
+	bs := []byte{}
```

Comment:
I imagine this still involves a heap allocation; that would be wasted effort if `b.len > 0`. Can you make this an `else` block on the `if` below?